src/Microsoft.AspNetCore.Mvc.Razor.Precompilation.Tools/Internal/PrecompileDispatchCommand.cs

```diff
@@ -174,69 +134,31 @@ private OutputPaths GetOutputPaths(IEnumerable<string> runtimeIdentifiers)
                 throw new InvalidOperationException($"Project '{ProjectPath}' does not support framework: {FrameworkOption.Value()}");
             }
 
-            var runtimeContext = workspace.GetRuntimeContext(projectContext, runtimeIdentifiers);
-            return runtimeContext.GetOutputPaths(
-                Configuration,
-                BuildBasePathOption.Value(),
-                OutputPathOption.Value());
-        }
+            var runtimeContext = workspace.GetRuntimeContext(
+                projectContext,
+                RuntimeEnvironmentRidExtensions.GetAllCandidateRuntimeIdentifiers());
 
-        private int BuildProject()
-        {
-            var arguments = new List<string>
-            {
-                ProjectPath,
-                "--framework",
-                TargetFramework.ToString(),
-                "--configuration",
-                Configuration,
-            };
-
-            if (BuildBasePathOption.HasValue())
-            {
-                arguments.Add("--build-base-path");
-                arguments.Add(BuildBasePathOption.Value());
-            }
-
-            if (OutputPathOption.HasValue())
-            {
-                arguments.Add("--output");
-                arguments.Add(OutputPathOption.Value());
-            }
-
-            return Command.CreateDotNet("build", arguments, TargetFramework, Configuration)
-                .ForwardStdErr(Console.Error)
-                .ForwardStdOut(Console.Out)
-                .Execute()
-                .ExitCode;
+            return runtimeContext.GetOutputPaths(Configuration);
         }
 
         private NuGetFramework GetTargetFramework()
         {
             if (!FrameworkOption.HasValue())
             {
-                var workspace = new BuildWorkspace(ProjectReaderSettings.ReadFromEnvironment());
-                var projectContexts = workspace.GetProjectContextCollection(ProjectPath)
-                    .FrameworkOnlyContexts
-                    .ToList();
-
-                if (projectContexts.Count == 0)
-                {
-                    throw new Exception($"Error: Project at {ProjectPath} does not have any specified frameworks.");
+                throw new Exception($"Option {FrameworkOption.Template} does not have a value.");
+            }
 
-                }
-                else if (projectContexts.Count > 1)
-                {
-                    throw new Exception($"Error: Project at {ProjectPath} supports multiple framework. " +
-                        $"Specify a framework using {FrameworkOption.Template}.");
-                }
+            return NuGetFramework.Parse(FrameworkOption.Value());
+        }
 
-                return projectContexts[0].TargetFramework;
-            }
-            else
+        private string GetOutputPath()
```

Comment:
Would prefer to stay consistent with the Design pattern where you have a single ParseArguments that validates the inputs to the application.